### PR TITLE
Add more tests for CopyStruct

### PR DESCRIPTION
**Pull request by nickpetrovic**

### CODE REVIEW

File: pkg/common/redis_test.go [hunk 1]
```diff
@@ -87,12 +87,133 @@ func TestRedisLockWithTTLAndRetry(t *testing.T) {
...
+				Status:       types.WorkerStatusPending,
+				TotalCpu:     20,
+				TotalMemory:  0,
+				BuildVersion: "0.1.0",
+			},
+			expected: &types.Worker{
+				Id:                   "123",
+				Status:               types.WorkerStatusAvailable,
+				TotalCpu:             10,
+				TotalMemory:          0,
```
Comment (luke-lombardi) [inline pkg/common/redis_test.go:145]: nit-pick but i think it would be worth making all of these non-zero values

**Review by luke-lombardi — CHANGES_REQUESTED** [review]

File: pkg/common/redis_test.go [hunk 2]
```diff
@@ -87,12 +87,133 @@ func TestRedisLockWithTTLAndRetry(t *testing.T) {
...
+			},
+		},
+		{
+			name: "copy struct overwriting fields on destination part 2",
+			src: &types.Worker{
+				Id:       "123",
+				Status:   types.WorkerStatusAvailable,
+				TotalCpu: 10,
+			},
+			dest: &types.Worker{
```
Comment (luke-lombardi) [inline pkg/common/redis_test.go:134]: @nickpetrovic can you add a test for machine state as well

File: pkg/common/redis_test.go (hunk 1 above)
Comment (dleviminzi) [reply pkg/common/redis_test.go:145]: This is a good library if you want to automate creating a bunch of fake structs a bit: https://github.com/go-faker/faker

Comment (nickpetrovic) [reply pkg/common/redis_test.go]: Yeah good call. The test after this tests for defaults. I've updated this.

File: pkg/common/redis_test.go (hunk 2 above)
Comment (nickpetrovic) [reply pkg/common/redis_test.go:146]: Added.

**Pull request merged**